tests/sentry/processing/realtime_metrics/test_redis.py

```diff
@@ -487,82 +519,117 @@ def test_get_durations_for_project_empty(
 ) -> None:
-    redis_cluster.hset("symbolicate_event_low_priority:histogram:10:42:-111", 0, 123)
+    redis_cluster.hset("symbolicate_event_low_priority:histogram:10:42:-110", 0, 123)
 
+    mock_time.return_value = -103.0
     counts = store.get_durations_for_project(42)
+    histogram = empty_histogram
+    histogram[0] = 123
+
     assert list(counts) == [
-        DurationHistogram(timestamp=-111, histogram=BucketedDurations({0: 123}))
+        DurationHistogram(timestamp=-110, histogram=BucketedDurations(histogram))
     ]
 
 
 def test_get_durations_for_project_negative_duration(
-    store: RedisRealtimeMetricsStore, redis_cluster: redis._RedisCluster
+    store: RedisRealtimeMetricsStore, redis_cluster: redis._RedisCluster, mock_time, empty_histogram
 ) -> None:
-    redis_cluster.hset("symbolicate_event_low_priority:histogram:10:42:111", -20, 123)
+    mock_time.return_value = 113.0
+    redis_cluster.hset("symbolicate_event_low_priority:histogram:10:42:110", -20, 123)
 
+    histogram = empty_histogram
+    histogram[-20] = 123
     counts = store.get_durations_for_project(42)
+
     assert list(counts) == [
-        DurationHistogram(timestamp=111, histogram=BucketedDurations({-20: 123}))
+        DurationHistogram(timestamp=110, histogram=BucketedDurations(histogram))
     ]
 
 
 def test_get_durations_for_project_negative_count(
-    store: RedisRealtimeMetricsStore, redis_cluster: redis._RedisCluster
+    store: RedisRealtimeMetricsStore, redis_cluster: redis._RedisCluster, mock_time, empty_histogram
 ) -> None:
-    redis_cluster.hset("symbolicate_event_low_priority:histogram:10:42:111", 0, -123)
+    mock_time.return_value = 113.0
+    redis_cluster.hset("symbolicate_event_low_priority:histogram:10:42:110", 0, -123)
 
     counts = store.get_durations_for_project(42)
+    histogram = empty_histogram
+    histogram[0] = -123
+
     assert list(counts) == [
-        DurationHistogram(timestamp=111, histogram=BucketedDurations({0: -123}))
+        DurationHistogram(timestamp=110, histogram=BucketedDurations(histogram))
     ]
 
 
 def test_get_durations_for_project_multi_key_multi_durations(
-    store: RedisRealtimeMetricsStore, redis_cluster: redis._RedisCluster
+    store: RedisRealtimeMetricsStore, redis_cluster: redis._RedisCluster, mock_time, empty_histogram
 ) -> None:
-    redis_cluster.hset("symbolicate_event_low_priority:histogram:10:42:111", 0, 123)
-    redis_cluster.hset("symbolicate_event_low_priority:histogram:10:42:111", 10, 456)
-    redis_cluster.hset("symbolicate_event_low_priority:histogram:10:42:222", 20, 123)
-    redis_cluster.hset("symbolicate_event_low_priority:histogram:10:53:111", 0, 123)
+    redis_cluster.hset("symbolicate_event_low_priority:histogram:10:42:110", 0, 123)
+    redis_cluster.hset("symbolicate_event_low_priority:histogram:10:42:110", 10, 456)
+    redis_cluster.hset("symbolicate_event_low_priority:histogram:10:53:110", 0, 123)
```

Comment:
similar to the suggestion for counters, could some testing be added to ensure that the hole-filling is happening between two different hash entries, as well as inside of a hash value? for example, this test can be altered slightly so the second `hset` uses a duration of `30` instead of `10`, and then the expected histogram can be extended accordingly. a separate test that performs this works as well, if you'd like to avoid bundling too many expectations into one test.

a separate test mimicking the counter test requested above where two histograms are initialized at 110 and 150 seconds could also neatly test the hole-filling behaviour between two hash entries.

Reply:
ah, i missed `empty_histogram`. in that case, if you feel that testing gaps between entries is redundant given that this test is already checking for zero-filling _after_ the populated entries, then feel free to skip that, and the appropriate sections in the first paragraph of my previous comment. the second comment i believe still holds, though